src/main/java/it/gov/pagopa/payhub/activities/service/exportflow/debtposition/PaidExportFlowFileService.java

```diff
@@ -0,0 +1,105 @@
+
+@Service
+@Lazy
+@Slf4j
+public class PaidExportFlowFileService extends BaseExportFlowFileService<PaidExportFile, PaidExportFileFilter,InstallmentPaidViewDTO, PaidInstallmentExportFlowFileDTO>{
+
+    private final int pageSize;
+    private final ExportFileService exportFileService;
+    private final DataExportService dataExportService;
+    private final InstallmentExportFlowFileDTOMapper installmentExportFlowFileDTOMapper;
+
+
+
+    public PaidExportFlowFileService(CsvService csvService,
+                                     Class<PaidInstallmentExportFlowFileDTO> csvRowDtoClass,
+                                     FileArchiverService fileArchiverService,
+                                     @Value("${folders.tmp}") Path workingDirectory,
+                                     @Value("${export-flow-files.paid.relative-file-folder}")String relativeFileFolder,
+                                     @Value("${export-flow-files.paid.filename-prefix}")String filenamePrefix,
+                                     @Value("${export-flow-files.paid.page-size}") int pageSize,
+                                     ExportFileService exportFileService,
+                                     DataExportService dataExportService,
+                                     InstallmentExportFlowFileDTOMapper installmentExportFlowFileDTOMapper
+                                     ) {
+
+        super(csvService, csvRowDtoClass, fileArchiverService, workingDirectory, relativeFileFolder, filenamePrefix);
+        this.pageSize = pageSize;
+        this.exportFileService = exportFileService;
+        this.dataExportService = dataExportService;
+        this.installmentExportFlowFileDTOMapper = installmentExportFlowFileDTOMapper;
+    }
+
+    @Override
+    public PaidExportFile findExportFileRecord(Long exportFileId) {
+        return exportFileService.findPaidExportFileById(exportFileId)
+                .orElseThrow(() -> new ExportFlowFileNotFoundException("Cannot found paidExportFile having id: %d".formatted(exportFileId)));
+    }
+
+    @Override
+    public Long getOrganizationId(PaidExportFile exportFile) {
+        return exportFile.getOrganizationId();
+    }
+
+    @Override
+    protected String getFlowFileVersion(PaidExportFile exportFile) {
+        return exportFile.getFlowFileVersion();
+    }
+
+    @Override
+    public List<InstallmentPaidViewDTO> retrievePage(PaidExportFile exportFile, PaidExportFileFilter filter, int pageNumber) {
+        List<InstallmentPaidViewDTO> installmentPaidViewDTOList = Collections.emptyList();
+
+        PagedInstallmentsPaidView pagedInstallmentsPaidView = dataExportService.exportPaidInstallments(exportFile.getOrganizationId(), exportFile.getOperatorExternalId(), filter, pageNumber, pageSize, List.of("installmentId"));
+
+        if (pagedInstallmentsPaidView != null){
+            installmentPaidViewDTOList= pagedInstallmentsPaidView.getContent();
+        }
+
+        return installmentPaidViewDTOList;
```

Comment:
I'm not a fun of such a pattern
```suggestion
        PagedInstallmentsPaidView pagedInstallmentsPaidView = dataExportService.exportPaidInstallments(exportFile.getOrganizationId(), exportFile.getOperatorExternalId(), filter, pageNumber, pageSize, List.of("installmentId"));

        if (pagedInstallmentsPaidView != null){
            return pagedInstallmentsPaidView.getContent();
        } else {
           return Collections.emptyList();
        }
```